communication/mavlink_communication.hpp

```diff
@@ -104,139 +91,79 @@ class Mavlink_communication
      */
-    Onboard_parameters& onboard_parameters();
-    Onboard_parameters* p_onboard_parameters();
-
-    /**
-     * \brief   Run task scheduler update if the buffer is empty
-     *
-     * \param   mavlink_communication   Pointer to the MAVLink communication structure
-     *
-     * \return  Task status return
-     */
-    static bool update(Mavlink_communication* mavlink_communication);
+    Onboard_parameters& parameters();
+    Onboard_parameters* p_parameters();
 
-private:
 
     /**
-     * \brief   MAVLink message handler structure
+     * \brief   Main update function
+     *
+     * \return  Success
      */
-    struct send_msg_handler_t
-    {
-        Mavlink_stream* mavlink_stream;                                 ///<    Pointer to the MAVLink stream structure
-        send_msg_function_t function;                                   ///<    Pointer to the function to be executed
-        handling_telemetry_module_struct_t      module_struct;          ///<    Pointer to module data structure to be given as argument to the function
-    };
-
-    Scheduler                   scheduler_;                      ///<    Task set for scheduling of down messages
-    Mavlink_stream              mavlink_stream_;                 ///<    Mavlink interface using streams
-    Mavlink_message_handler     message_handler_;                ///<    Message handler
-    Onboard_parameters          onboard_parameters_;             ///<    Onboard parameters
-    uint32_t                    msg_sending_count_;                                     ///<    Number of message callback currently registered
-    uint32_t                    max_msg_sending_count_;                                 ///<    Maximum number of callback that can be registered
-    send_msg_handler_t*         msg_send_list_;                      ///<    List of message callbacks
+    bool update(void);
 
 
     /**
-     * \brief   Suspending sending of messages
+     * \brief   Main update function, static version
      *
-     * \param   msg_send    The MAVLink message sending handler
-     *
-     * \return  The result of execution of the task
-     */
-    static bool send_message(send_msg_handler_t* msg_send);
-
-    /**
-     * \brief   Toggle mavlink telemetry stream
+     * \param   mavlink_communication   Pointer to the MAVLink communication structure
      *
-     * \param scheduler     scheduler of the MAV
-     * \param sysid         MAV sysid
-     * \param msg           pointer to the stream you want to toggle
+     * \return  Success
      */
-    static void toggle_telemetry_stream(Scheduler* scheduler, uint32_t sysid, mavlink_message_t* msg);
-
+    static bool update_task(Mavlink_communication* mavlink_communication);
 
+private:
+    Mavlink_stream                      mavlink_stream_;       ///<    Mavlink interface using streams
+    Periodic_telemetry_tpl<30>          telemetry_;            ///<    Periodic telemetry
```

Comment:
done in https://github.com/lis-epfl/MAVRIC_Library/pull/326/commits/eba4b84c199b32b2f0cf368c85b62f0851c6b0db